modules/installation-user-infra-machines-advanced-enabling-serial-console.adoc

```diff
@@ -0,0 +1,44 @@
+// Module included in the following assemblies:
+//
+// * installing/installing_bare_metal/installing-bare-metal.adoc
+// * installing/installing_bare_metal/installing-restricted-networks-bare-metal.adoc
+// * installing/installing_bare_metal/installing-bare-metal-network-customizations.adoc
+
+:_content-type: PROCEDURE
+[id="installation-user-infra-machines-advanced-enabling-serial-console_{context}"]
+= Enabling the serial console for PXE and ISO installations
+
+By default, the serial console is disabled and all output is written to the graphical console. The following procedure for an ISO installation enables the serial console and reconfigures the bootloader so that output is sent to both the serial console and the graphical console.
+
+.Procedure
+
+. Boot the ISO installer.
+
+. Run the `coreos-installer` command to install the system, adding the `--console` option once to specify the graphical console, and a second time to specify the serial console:
++
+[source,terminal]
+----
+$ coreos-installer install --console=tty0 --console=ttyS0 \
+  --ignition-url=http://host/worker.ign /dev/sda
+----
+
+. Reboot into the installed system.
++
+[NOTE]
+====
+A similar outcome can be obtained by using the `coreos-installer install --append-karg` option, and specifying the console with `console=`. However, this will only set the console for the kernel and not the bootloader.
+====
+
+To configure a PXE installation, you must add `console=` arguments to the `APPEND` line of your menu entry. For example, consider the following menu entry for `x86_64` architectures:
+
+[source,text]
+----
+DEFAULT pxeboot
+TIMEOUT 20
+PROMPT 0
+LABEL pxeboot
+    KERNEL http://<HTTP_server>/rhcos-<version>-live-kernel-<architecture>
+    APPEND initrd=http://<HTTP_server>/rhcos-<version>-live-initramfs.<architecture>.img coreos.live.rootfs_url=http://<HTTP_server>/rhcos-<version>-live-rootfs.<architecture>.img console=tty0 console=ttyS0 coreos.inst.install_dev=/dev/sda coreos.inst.ignition_url=http://<HTTP_server>/bootstrap.ign
+----
+
+The arguments `console=tty0 console=ttyS0` set the serial console as the primary console and the graphical console as a secondary console.
```

Comment:
Those arguments are not automatically applied to the installed system, so this only affects the live system and not the installed one.

Reply:
How would a user apply these to the installed system? Would it make sense to link out to [this procedure](https://docs.openshift.com/container-platform/4.11/installing/installing_bare_metal/installing-bare-metal.html#installation-user-infra-machines-pxe_installing-bare-metal) in some way?

Reply:
If they're PXE booting without `coreos.inst.install_dev`, they'll get a shell prompt and they can run `coreos-installer` manually, just as in the ISO case.  If they want to automate the install, the only way to configure the destination console is with a custom image created with `coreos-installer pxe customize --dest-console`.  That's intentional; we're trying to steer advanced customization toward `iso/pxe customize` and away from kernel arguments.